safe/gui/tools/wizard/test/test_function_centric_wizard.py

```diff
@@ -0,0 +1,897 @@
+# coding=utf-8
+"""
+InaSAFE Disaster risk assessment tool developed by AusAid and World Bank
+- **GUI Test Cases.**
+
+Contact : ole.moller.nielsen@gmail.com
+
+.. note:: This program is free software; you can redistribute it and/or modify
+     it under the terms of the GNU General Public License as published by
+     the Free Software Foundation; either version 2 of the License, or
+     (at your option) any later version.
+
+"""
+
+__author__ = 'borysjurgiel.pl'
+__date__ = '24/02/2014'
+__copyright__ = ('Copyright 2012, Australia Indonesia Facility for '
+                 'Disaster Reduction')
+
+import unittest
+import sys
+import os
+import shutil
+
+# Import qgis in order to set SIP API.
+# pylint: disable=unused-import
+import qgis
+# pylint: enable=unused-import
+from qgis.core import QgsMapLayerRegistry
+from PyQt4 import QtCore
+# noinspection PyPackageRequirements
+from PyQt4.QtCore import Qt
+
+# noinspection PyPackageRequirements
+# Add PARENT directory to path to make test aware of other modules
+pardir = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), '../../..///'))
+sys.path.append(pardir)
+
+from safe.common.utilities import temp_dir
+from safe.test.utilities import (
+    clone_raster_layer,
+    clone_shp_layer,
+    get_qgis_app,
+    test_data_path)
+
+# AG: get_qgis_app() should be called before importing modules from
+# safe.gui.tools.wizard
+QGIS_APP, CANVAS, IFACE, PARENT = get_qgis_app()
+
+from safe.definitions import inasafe_keyword_version
+from safe.impact_functions import register_impact_functions
+from safe.gui.tools.wizard.wizard_dialog import WizardDialog
+from safe.gui.widgets.dock import Dock
+from safe.utilities.keyword_io import KeywordIO, definition
+
+DOCK = Dock(IFACE)
+
+
+# noinspection PyTypeChecker
+class WizardDialogTest(unittest.TestCase):
+    """Test the InaSAFE wizard GUI"""
+    def setUp(self):
+        # register impact functions
+        register_impact_functions()
+
+    def tearDown(self):
+        """Run after each test."""
+        # Remove the mess that we made on each test
+        shutil.rmtree(temp_dir(sub_dir='test'))
+
+    def check_current_step(self, expected_step):
+        """Helper function to check the current step is expected_step
+
+        :param expected_step: The expected current step.
+        :type expected_step: WizardStep instance
+        """
+        current_step = expected_step.parent.get_current_step()
+        message = ('Expected %s but I got %s' % (expected_step, current_step))
+        self.assertEqual(expected_step, current_step, message)
```

Comment:
I think we can use the default failed assert message, rather than creating our own here.